Code/GraphMol/ScaffoldNetwork/ScaffoldNetwork.h

```diff
@@ -0,0 +1,177 @@
+//
+//  Copyright (C) 2019 Greg Landrum and T5 Informatics GmbH
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <RDGeneral/export.h>
+#ifndef RD_SCAFFOLDNETWORK_H
+#define RD_SCAFFOLDNETWORK_H
+
+#include <vector>
+#include <map>
+#include <string>
+#include <sstream>
+#include <memory>
+#include <iostream>
+
+#ifdef RDK_USE_BOOST_SERIALIZATION
+#include <RDGeneral/Invariant.h>
+#include <RDGeneral/BoostStartInclude.h>
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/serialization/vector.hpp>
+#include <boost/serialization/shared_ptr.hpp>
+#include <RDGeneral/BoostEndInclude.h>
+#endif
+
+namespace RDKit {
+class ROMol;
+class ChemicalReaction;
+
+namespace ScaffoldNetwork {
+
+struct RDKIT_SCAFFOLDNETWORK_EXPORT ScaffoldNetworkParams {
+  bool includeGenericScaffolds =
```

Comment:
Have I mentioned how much I like this intializer style?

Reply:
yeah, it's great